### PR TITLE
Email service for Submission notifications

### PR DESCRIPTION
**Pull request by parc-jason**
# Description
Add email service to deliver emails for submission confirmation (submitter enters email and POSTS) and for submission received (emails go to configured staff).

Added new endpoint POST /submissions/ID/email body = {to: <email submitter fills out>}

Email service will throw errors, up to the caller how it is handled.

Expected that calling an email service, the permissions have already been checked, so this service will do no current user permission checking.

**IMPORTANT** since there is no persisted configuration for emails, I am using the request header referrer to determine the links in the email.  This does work on OpenShift and localhost, but using Postman you will need to set this header.

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-wxpbtr-tools.pathfinder.gov.bc.ca/dashboard?id=common-hosted-form-service-pr-90

File: app/src/forms/email/emailService.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
+const chesService = require('../../components/chesService');
+const fs = require('fs');
+const log = require('npmlog');
+const path = require('path');
+
+const formService = require('../form/service');
+
+const service = {
+
+  _appUrl: (referer) => {
```
Comment (jujaga) [inline app/src/forms/email/emailService.js:10]: Typo: referrer, not referer (happens in multiple places)

File: app/src/forms/form/controller.js [hunk 2]
```diff
@@ -104,6 +105,7 @@ module.exports = {
   createSubmission:  async (req, res, next) => {
     try {
       const response = await service.createSubmission(req.params.formVersionId, req.body, req.currentUser);
+      emailService.submissionReceived(req.params.formId, response.id, req.headers.referer).catch(() => {});
```
Comment (jujaga) [inline app/src/forms/form/controller.js:108]: Typo - referrer, not referer. Definitely need to make sure we're not forcing API calls to use the wrong spelling.

File: app/src/forms/submission/controller.js [hunk 3]
```diff
@@ -16,6 +17,15 @@ module.exports = {
     } catch (error) {
       next(error);
     }
+  },
+  email: async (req, res, next) => {
+    try {
+      const submission = await service.read(req.params.formSubmissionId, req.currentUser);
+      const response = await emailService.submissionConfirmation(submission.form.id, req.params.formSubmissionId, req.body, req.headers.referer);
```
Comment (jujaga) [inline app/src/forms/submission/controller.js:24]: Typo - referrer, not referer. Definitely need to make sure we're not forcing API calls to use the wrong spelling.

**Review by jujaga — CHANGES_REQUESTED** [review]
Outside of the typo, looks pretty good. Requesting a change on this as the typo affects the exposed API.

File: app/src/forms/email/emailService.js (hunk 1 above)
Comment (parc-jason) [reply app/src/forms/email/emailService.js:10]: https://en.wikipedia.org/wiki/HTTP_referer

**Review by jujaga — APPROVED** [review]
TIL RFC 1945 has "referer" deliberately spelt like that. 👍

**Pull request merged**